reader.go

```diff
@@ -33,6 +34,16 @@ func NewReader(r io.ReadSeeker) Reader {
 	return &reader{r: r}
 }
 
+// NewReaderWithOffset creates a new Reader giving the position of the index.
+// This is useful to open siva files that are being written or reading an
+// old index.
+func NewReaderWithOffset(r io.ReadSeeker, o uint64) Reader {
```

Comment:
what happens if the provided offset is not an index?

Reply:
Calling `Index` function will error. The same way it errors when opening a broken siva file.